regional_mom6/regional_mom6.py

```diff
@@ -1678,52 +2317,163 @@ def setup_run_directory(
             # This prevents the layout from being overwritten in the main class in case
             # in case the user accidentally loads in the wrong mask table.
             layout = self.layout
-            ncpus = layout[0] * layout[1]
+            if layout == None:
+                print(
+                    "WARNING: No mask table found, and the cpu layout has not been set. \nAt least one of these is requiret to set up the experiment if you're running MOM6 standalone with the FMS coupler. \nIf you're running within CESM, ignore this message."
+                )
+            else:
+                print(
+                    f"No mask table found, but the cpu layout has been set to {self.layout} This suggests the domain is mostly water, so there are "
+                    + "no `non compute` cells that are entirely land. If this doesn't seem right, "
+                    + "ensure you've already run the `FRE_tools` method which sets up the cpu mask table. Keep an eye on any errors that might print while"
+                    + "the FRE tools (which run C++ in the background) are running."
+                )
 
-        print("Number of CPUs required: ", ncpus)
+                ncpus = layout[0] * layout[1]
+                print("Number of CPUs required: ", ncpus)
 
-        ## Modify the input namelists to give the correct layouts
+        ## Modify the MOM_layout file to have correct horizontal dimensions and CPU layout
         # TODO Re-implement with package that works for this file type? or at least tidy up code
-        with open(self.mom_run_dir / "MOM_layout", "r") as file:
-            lines = file.readlines()
-            for jj in range(len(lines)):
-                if "MASKTABLE" in lines[jj]:
-                    if mask_table != None:
-                        lines[jj] = f'MASKTABLE = "{mask_table}"\n'
-                    else:
-                        lines[jj] = "# MASKTABLE = no mask table"
-                if "LAYOUT =" in lines[jj] and "IO" not in lines[jj]:
-                    lines[jj] = f"LAYOUT = {layout[1]},{layout[0]}\n"
+        MOM_layout_dict = self.read_MOM_file_as_dict("MOM_layout")
+        if "MASKTABLE" in MOM_layout_dict.keys():
+            if mask_table != None:
+                MOM_layout_dict["MASKTABLE"]["value"] = mask_table
+            else:
+                MOM_layout_dict["MASKTABLE"]["value"] = "# MASKTABLE = no mask table"
+        if (
+            "LAYOUT" in MOM_layout_dict.keys()
+            and "IO" not in MOM_layout_dict.keys()
+            and layout != None
+        ):
+            MOM_layout_dict["LAYOUT"]["value"] = str(layout[1]) + "," + str(layout[0])
+        if "NIGLOBAL" in MOM_layout_dict.keys():
+            MOM_layout_dict["NIGLOBAL"]["value"] = self.hgrid.nx.shape[0] // 2
+        if "NJGLOBAL" in MOM_layout_dict.keys():
+            MOM_layout_dict["NJGLOBAL"]["value"] = self.hgrid.ny.shape[0] // 2
```

Comment:
Not sure why `IO` shouldn't exist in the first condition?

```suggestion
        if (
            "LAYOUT" in MOM_layout_dict
            and "IO" not in MOM_layout_dict
            and layout != None
        ):
            MOM_layout_dict["LAYOUT"]["value"] = str(layout[1]) + "," + str(layout[0])
        if "NIGLOBAL" in MOM_layout_dict:
            MOM_layout_dict["NIGLOBAL"]["value"] = self.hgrid.nx.shape[0] // 2
        if "NJGLOBAL" in MOM_layout_dict:
            MOM_layout_dict["NJGLOBAL"]["value"] = self.hgrid.ny.shape[0] // 2
```

Reply:
Thanks for this catch! In premade_run_directories, the field in MOM_layout is "IO_Layout". This existed in the code previously, but I just translated it here, I think incorrectly. I can change it to check for "IO_Layout". We don't use MOM_layout so I don't have a great idea what this means however. @ashjbarnes Any input?

Reply:
I think this fix is right, and it runs on COSIMA computers so I'll resolve